lib/function_object.rb

```diff
@@ -30,6 +28,13 @@ def curry(arity = 0)
       end
       self
     end
+
+  private
+
+    def include_macro_mixin(macro)
```

Comment:
Extracting this stuff would be good but I don't want to bring any methods other than API (even private ones) to the user's class. `FunctionObject` adds only aDSL for setting up argument list and also stuff like `to_proc`, `curry`, etc. That's all. I really want to pollute user's class as little as possible.
I don't even like including DSL methods to the class because what if user wants to have its own `args` class method with a different purpose? But for now it's a necessary evil and I don't know how to get rid of it :(

I propose a different approach: move this code to `Macro#apply` to use it like `Macros.new(arg_descs).apply(self)`. It makes a lot of sense to me since mixing in the `class_mixin` and `sclass_mixin` is the *implementation detail* of how macro is applied to the class. `Macro#class_mixin` and `Macros.sclass_mixin` should also be set `private`.